omod/src/main/java/org/openmrs/web/controller/user/UserFormController.java

```diff
@@ -125,6 +128,24 @@ public String showForm(@RequestParam(required = false, value = "userId") Integer
 			model.addAttribute("secretQuestion", userService.getSecretQuestion(user));
 		}
 		
+		if (user.getPerson().getId() != null && !Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty()) {
+			model.addAttribute("isProvider", true);
+			
+			//Since identifier can be null, check corresponding identifiers and replace the 'nulls' with a "no identifier"
+			List<Provider> providerList = new ArrayList<Provider>(Context.getProviderService().getProvidersByPerson(user.getPerson()));
+			for (int i = 0; i < providerList.size(); i++) {
+				if (providerList.get(i).getIdentifier() == null) {
+					Provider currentProvider = providerList.get(i);
+					currentProvider.setIdentifier("no identifier");
```

Comment:
This is actually bad practice, you don't want to set a new property value especially when you're dealing with hibernate managed objects because these are normally proxies, also **No identifier** needs to be translated, **No Identifier Specified** would probably be clearer. The proper thing to do is not set it here and in the JSP you check if it's not null you print the value otherwise display the 'No Identifier Specified' text, it would also be nice to put it in bold and italics so that one doesn't thing it is the actual identifier but rather a place holder